src/tools/perf/cuda/cuda_device_executor.c

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright (c) NVIDIA CORPORATION & AFFILIATES, 2025. ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#ifdef HAVE_CONFIG_H
+#  include "config.h"
+#endif
+
+#include <tools/perf/lib/libperf_int.h>
+
+#include <cuda_runtime.h>
+
+
+static ucs_status_t ucx_perf_cuda_executor_init(ucx_perf_device_test_h *test,
+                                                ucx_perf_context_t *perf)
+{
+    return UCS_OK;
+}
+
+static ucs_status_t ucx_perf_cuda_executor_cleanup(ucx_perf_device_test_h test)
+{
+    return UCS_OK;
+}
+
+static ucs_status_t ucx_perf_cuda_executor_sender(ucx_perf_device_test_h test,
```

Comment:
do we really need init/cleanup?
also why need sender/receiver call?
i'd expect we only have single callback function - reuse ucp_dispatch_func_t / uct_dispatch_func_t